### PR TITLE
ENH: Add allow_pickle option for savez and savez_compressesd

### PR DESCRIPTION
**Pull request by tinaoberoi**
Reference: [Issue #13983 ](https://github.com/numpy/numpy/issues/13983)
Changes: 
- Edit function `savez `  and  ` savez_compressed `in `numpy/numpy/lib/npyio.py`
- Edit Documentation

### CODE REVIEW

File: numpy/lib/npyio.py [hunk 1]
```diff
@@ -535,7 +535,7 @@ def _savez_dispatcher(file, *args, **kwds):
 
 
 @array_function_dispatch(_savez_dispatcher)
-def savez(file, *args, **kwds):
+def savez(file, allow_pickle=True, *args, **kwds):
```
Comment (eric-wieser) [inline numpy/lib/npyio.py:538]:
```suggestion
def savez(file, *args, allow_pickle=True, **kwds):
```

**Comment by eric-wieser** [comment]
See my comments in the issue. This is unfortunately a breaking change, which makes me thing we should not do it.

File: numpy/lib/npyio.py [hunk 2]
```diff
@@ -622,7 +622,7 @@ def _savez_compressed_dispatcher(file, *args, **kwds):
 
 
 @array_function_dispatch(_savez_compressed_dispatcher)
-def savez_compressed(file, *args, **kwds):
+def savez_compressed(file, allow_pickle=True, *args, **kwds):
```
Comment (eric-wieser) [inline numpy/lib/npyio.py]:
```suggestion
def savez_compressed(file, *args, allow_pickle=True, **kwds):
```

Comment (tinaoberoi) [reply numpy/lib/npyio.py]: Done @eric-wieser

File: numpy/lib/npyio.py (hunk 1 above)
Comment (tinaoberoi) [reply numpy/lib/npyio.py]: Done @eric-wieser

File: numpy/lib/npyio.py [hunk 3]
```diff
@@ -613,7 +613,7 @@ def savez(file, *args, **kwds):
     >>> npzfile['x']
     array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
     """
-    _savez(file, args, kwds, False)
+    _savez(file, args, kwds, allow_pickle)
```
Comment (eric-wieser) [inline numpy/lib/npyio.py:616]:
```suggestion
    _savez(file, args, kwds, False, allow_pickle)
```

File: numpy/lib/npyio.py [hunk 4]
```diff
@@ -683,7 +683,7 @@ def savez_compressed(file, *args, **kwds):
     True
 
     """
-    _savez(file, args, kwds, True)
+    _savez(file, args, kwds, allow_pickle)
```
Comment (eric-wieser) [inline numpy/lib/npyio.py:686]:
```suggestion
    _savez(file, args, kwds, True, allow_pickle)
```

**Review by eric-wieser — CHANGES_REQUESTED** [review]
Just so this appears in the issue list - the PR looks like a reasonable interpretation of the issue, but I think the issue itself is a bad idea: https://github.com/numpy/numpy/issues/13983#issuecomment-632037925

**Comment by rossbar** [comment]
This was discussed at a triage meeting and the consensus was that the approach motivated by the linked issue was not desirable, so I will close this for now. Thanks @tinaoberoi !

**Pull request closed without merging**